lib/IMAP/MessageMapper.php

```diff
@@ -178,6 +178,20 @@ function (int $uid) use ($highestKnownUid) {
 				return $uid > $highestKnownUid;
 			}
 		);
+		if(count($uidCandidates) === 0) {
+			if($upper < $max) {
+				$this->logger->debug("Range for findAll did not find any (not already known) messages. Trying again with a succeeding range... $upper < $max");
+				return $this->findAll($client, $mailbox, $maxResults, $upper);
+			} else {
+				$this->logger->debug("Range for findAll did not find any (not already known) messages and all messages of mailbox $mailbox have been fetched.");
+				return [
+					'messages' => [],
+					'all' => true,
```

Comment:
If this happens in the middle of a sync, the sync would stop here and never fetch candidates with higher UIDs due to all being true.

Reply:
But this is in the branch where $upper already reached or exceeded $max, which I understand is the MAX UID returned by the IMAP server for this query - thus the last range of this query is empty.

The first branch on the other hand continues with the next range.

Reply:
Right, ignore my comment :sweat_smile:

I'm currently working on the same piece of code, so it might take some time to merge this just to be sure that there are no unintended side effects :slightly_smiling_face: